domain/model/service/modelservice.go

```diff
@@ -339,7 +348,12 @@ func (s *ProviderModelService) CreateModel(
 	ctx context.Context,
 	controllerUUID uuid.UUID,
 ) error {
-	if err := s.CreateModelForVersion(ctx, controllerUUID, agentVersionSelector()); err != nil {
+	if err := s.CreateModelForVersion(
+		ctx,
+		controllerUUID,
+		agentVersionSelector(),
```

Comment:
Agent version and stream are really a tuple. If we're using an agent version selector indirection, that function could return the version and stream together, not just version.

Reply:
Fair point.